pkg/pfconfig/pfconfig.go

```diff
@@ -0,0 +1,225 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pfconfig implements the JSON configuration of Semtech's reference Packet Forwarder.
+package pfconfig
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"go.thethings.network/lorawan-stack/pkg/errors"
+)
+
+// Config represents the full configuration for Semtech's reference Packet Forwarder.
+type Config struct {
+	SX1301Conf  SX1301Config `json:"SX1301_conf"`
```

Comment:
The GS needs only the `SX1301Conf` to construct the config for the BasicStation devices. 

So, currently, I need to build `Config` and discard the `GatewayConf`.

I feel it's better to extend this package to build the LNS config also. With that in mind, this could be called (needs better naming) "PFConfig" and the LNS one as "BSConfig" which would use only the `SX1301Conf`.

Reply:
See other comment